src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs

```diff
@@ -175,21 +184,27 @@ public Collection<AttachmentSet> AfterTestRunEnd(bool isCanceled, ITestMessageEv
         /// </summary>
         public void Dispose()
         {
+            EqtTrace.Info("ProxyDataCollectionManager.Dispose: calling dospose for datacollector processId: {0} port: {1}", this.dataCollectionProcessId, this.dataCollectionPort);
             this.dataCollectionRequestSender.Close();
         }
 
         /// <inheritdoc />
         public void Initialize()
         {
-            var port = this.dataCollectionRequestSender.InitializeCommunication();
+            this.dataCollectionPort = this.dataCollectionRequestSender.InitializeCommunication();
 
             // Warn the user that execution will wait for debugger attach.
-            var processId = this.dataCollectionLauncher.LaunchDataCollector(null, this.GetCommandLineArguments(port));
+            this.dataCollectionProcessId = this.dataCollectionLauncher.LaunchDataCollector(null, this.GetCommandLineArguments(this.dataCollectionPort));
+            EqtTrace.Info("ProxyDataCollectionManager.Initialize: Launched datacollector processId: {0} port: {1}", this.dataCollectionProcessId, this.dataCollectionPort);
+
+            ChangeConnectionTimeoutIfRequired(dataCollectionProcessId);
+
+            EqtTrace.Info("ProxyDataCollectionManager.Initialize: waiting for connection with timeout: {0}", this.connectionTimeout);
 
-            ChangeConnectionTimeoutIfRequired(processId);
             var connected = this.dataCollectionRequestSender.WaitForRequestHandlerConnection(this.connectionTimeout);
             if (connected == false)
             {
+                EqtTrace.Error("ProxyDataCollectionManager.Initialize: failed to connect to datacollector process.");
```

Comment:
Add datacollector process ID we could not connect to

Reply:
Done.